manubot/cite/csl_item.py

```diff
@@ -275,3 +301,75 @@ def standardize_id(self):
 def assert_csl_item_type(x):
     if not isinstance(x, CSL_Item):
         raise TypeError(f"Expected CSL_Item object, got {type(x)}")
+
+
+def date_to_date_parts(iso_date):
+    """
+    Convert a date string or object to a date parts list.
+
+    iso_date: date either as a string (in the form YYYY, YYYY-MM, or YYYY-MM-DD)
+        or as a Python date object (datetime.date or datetime.datetime).
+    """
+    import datetime
+
+    if iso_date is None:
+        return None
+    if isinstance(iso_date, (datetime.date, datetime.datetime)):
+        iso_date = iso_date.isoformat()
+    if not isinstance(iso_date, str):
+        raise ValueError(f"date_to_date_parts: unsupported type for {iso_date}")
+    iso_date = iso_date.strip()
+    re_year = r"(?P<year>[0-9]{4})"
+    re_month = r"(?P<month>1[0-2]|0[1-9])"
+    re_day = r"(?P<day>[0-3][0-9])"
```

Comment:
This could be a slightly narrower range, similar to how you excluded some double digit numbers from the month regex.  But do we actually need to check for valid dates here?

Reply:
Yeah the function does a limited and incomplete check for date validity. I am okay with this for now. We haven't really decided where date validity checking should go, if anywhere.